tests/CiIntegrationTest.php

```diff
@@ -0,0 +1,187 @@
+        static::executeCommand('git clean -fd', true);
+        static::executeCommand('git checkout master', true);
+    }
+
+    public function testIntegration()
+    {
+        static::executeCommand('git checkout case1', true);
+
+        $steps = array(
+            'COMMIT_RANGE="master..case1"',
+            file_get_contents(__DIR__.'/../dev-tools/ci-integration/step1-changed_files.sh'),
+            file_get_contents(__DIR__.'/../dev-tools/ci-integration/step2-extra_args.sh'),
+            str_replace('vendor/bin/', './../../../', file_get_contents(__DIR__.'/../dev-tools/ci-integration/step3-execution.sh')),
+        );
+
+        $result1 = static::executeScript(array(
+            $steps[0],
+            $steps[1],
+            'echo "${#CHANGED_FILES[@]}"',
+            'echo "${CHANGED_FILES[@]}"',
+            'echo "${CHANGED_FILES[0]}"',
+            'echo "${CHANGED_FILES[1]}"',
+        ), true);
+
+        $this->assertSame(
+            array(
+                '2',
+                'dir a/file.php dir b/file b.php',
+                'dir a/file.php',
+                'dir b/file b.php',
+            ),
+            explode("\n", rtrim($result1['output']))
+        );
+
+        $result2 = static::executeScript(array(
+            $steps[0],
+            $steps[1],
+            $steps[2],
+            'echo "${#EXTRA_ARGS[@]}"',
+            'echo "${EXTRA_ARGS[@]}"',
+            'echo "${EXTRA_ARGS[0]}"',
+            'echo "${EXTRA_ARGS[1]}"',
+            'echo "${EXTRA_ARGS[2]}"',
+            'echo "${EXTRA_ARGS[3]}"',
+        ), true);
+
+        $this->assertSame(
+            array(
+                '4',
+                '--path-mode=intersection -- dir a/file.php dir b/file b.php',
+                '--path-mode=intersection',
+                '--',
+                'dir a/file.php',
+                'dir b/file b.php',
+            ),
+            explode("\n", rtrim($result2['output']))
+        );
+
+        $result3 = static::executeScript(array(
+            $steps[0],
+            $steps[1],
+            $steps[2],
+            $steps[3],
+        ), true);
+
+        $optionalIncompatibilityWarning = 'PHP needs to be a minimum version of PHP 5.3.6 and maximum version of PHP 7.1.*.
+Ignoring environment requirements because `PHP_CS_FIXER_IGNORE_ENV` is set. Execution may be unstable.
+';
+        $executionDetails = 'Loaded config default from ".php_cs.dist".
+S.
+Legend: ?-unknown, I-invalid file syntax, file ignored, S-Skipped, .-no changes, F-fixed, E-error';
+
+        $this->assertRegExp(
+            '/^('.preg_quote($optionalIncompatibilityWarning, '/').')?'.preg_quote($executionDetails, '/').'$/',
+            trim($result3['stderr'])
+        );
+        $this->assertRegExp(
+            '/^Checked all files in \d+\.\d+ seconds, \d+\.\d+ MB memory used$/',
+            trim($result3['output'])
+        );
+        $this->assertSame(0, $result3['code']);
```

Comment:
This assertion is not needed because the related `static::executeScript()` call will throw an exception if the exit code of the command is not 0.

I would drop the `$crashOnError` parameter from `executeScript()`, make it always pass `$crashOnError = false` to `executeCommand()` and check the exit code of each command *before* checking output and stderr.